docs/_drafts/go-strategy-pattern-a-response.md

```diff
@@ -0,0 +1,172 @@
+If you're dealing with `int32`s, you may choose radix-sort; if the list is likely to be almost sorted, you might prefer insertion-sort to quick-sort.
+
+There are three benefits to using the Strategy Pattern.
+Firstly, it decouples the calling logic from the strategy logic, which provides loose coupling and generality, and allows the implementation of incidental logic to be swapped out, such as for testing purposes.
+If the calling context doesn't need to know _how_ something is achieved, but just that it is, then it doesn't need to take responsilbity for that or concern itself with configuration for this; this aligns well with the Single Responsibility Principle.
+Secondly, it provides extensibility in line with the Open-Closed Principle, which is about keeping code more maintainable.
+Thirdly, it keeps the logic grouped into smaller, more focused elements rather than lengthy, expansive blocks.
+That should make it more legible and thus more maintainable, particularly as it's capturing the _purpose_ of these strategies and not just their implementations.[1]
+
+## OOPs!  Go is classless
+
+The description of the Strategy Pattern given above talks about classes, but Go doesn't have classes so how does it apply?!
+Redowan mentions that "the Go community exhibits a knee-jerk reaction to the word 'pattern'" but, rightly in my opinion, continues to use the established term rather than inventing a new one.
+
+What I think is missed here is that Go isn't a million miles away from traditional object-oriented (OO) languages, particularly given the trend of favouring composition over inheritance.
+Crucially, Go supports methods on structs rather than forcing the use of functions without receivers (`func (a *A) foo()` vs `func foo(a *A)`) --- **structs with methods are analogous to classes without subtype polymorphism** (inheritance)!
+Like many OO languages it has interfaces and, since Go 1.18, generics.
+Go is a memory-managed, garbage-collected language just like Java, C#, and Python, and its structs can be stack- or heap-allocated like in C++ and C#[2].
+Go's pointers are similar to references in languages like Java in that they do not support pointer arithmetic, although they do require manual indirection (`&` and `*`).
+Similar to Python's handling of visibility modifiers, there's public and private state; this is called exported and unexported state in Go nomenclature.
+
+In short, if you take a classic OO language and strictly forbid inheritance in favour of composition, it's not necessarily going to look too different to Go.
+As a consequence, I'd argue the use of design patterns, even ones common in OOP, are still valid and applicable.
+
+## Unwrapping unnecessary complexity
+
+My primary complaint with the rednafi's article is that the approach it presents is overcomplicated and, in my opinion, not particularly idiomatic.
```

Comment:
```suggestion
My primary complaint with rednafi's article is that the approach it presents is overcomplicated and, in my opinion, not particularly idiomatic.
```

Reply:
Fixed in https://github.com/agrski/agrski.github.io/pull/45/commits/1bf42f6f0cb97c9d730adf23fc54d223f6592959